lxd/storage/drivers/utils_cgo.go

```diff
@@ -106,7 +106,7 @@ static int find_associated_loop_device(const char *loop_file,
 			continue;
 
 		// Open fd to loop device.
-		return open(loop_dev_name, O_RDWR);
+		return open(loop_dev_name, O_RDWR | O_CLOEXEC);
```

Comment:
Very odd.

Reply:
See https://trello.com/c/SMSaDqDE/853-investigate-intermittent-network-test-failures#comment-60f96705485ef55c658f3807 for how it manifested.

Reply:
Actually I'll paste it here for posterity:

```
lxc stop c1-macvlan
Error: Error deleting storage volume: Failed to run: zfs destroy -r default/containers/c1-macvlan: umount: /var/snap/lxd/common/lxd/storage-pools/default/containers/c1-macvlan: target is busy.
cannot unmount '/var/snap/lxd/common/lxd/storage-pools/default/containers/c1-macvlan': umount failed

sudo nsenter --mount=/run/snapd/ns/lxd.mnt -- /var/lib/snapd/hostfs/usr/bin/lsof +D /var/snap/lxd/common/lxd/storage-pools/default/containers/c1-macvlan
COMMAND      PID USER   FD   TYPE DEVICE SIZE/OFF  NODE NAME
qemu-syst 119666  lxd   42u   REG   0,79      114 22045 /var/snap/lxd/common/lxd/storage-pools/default/containers/c1-macvlan/rootfs/var/lib/dpkg/info/libbrotli1:amd64.shlibs

ps aux | grep qemu-system | grep 119666
lxd       119666  3.8  0.8 1877084 1101624 ?     SLl  12:25   0:30 /snap/lxd/21081/bin/qemu-system-x86_64 -S -name v1-physical -uuid b55dd4c1-7d79-4e76-8287-db7537860b23 -daemonize -cpu host -nographic -serial chardev:console -nodefaults -no-user-config -sandbox on,obsolete=deny,elevateprivileges=allow,spawn=deny,resourcecontrol=deny -readconfig /var/snap/lxd/common/lxd/logs/v1-physical/qemu.conf -spice unix=on,disable-ticketing=on,addr=/var/snap/lxd/common/lxd/logs/v1-physical/qemu.spice -pidfile /var/snap/lxd/common/lxd/logs/v1-physical/qemu.pid -D /var/snap/lxd/common/lxd/logs/v1-physical/qemu.log -smbios type=2,manufacturer=Canonical Ltd.,product=LXD -runas 

lxc stop v1-physical
lxc delete c1-macvlan # completes OK now
```